src/UI/Shared/WB.UI.Shared.Enumerator/Activities/CompleteInterviewFragment.cs

```diff
@@ -13,73 +29,290 @@ public class CompleteInterviewFragment : BaseFragment<CompleteInterviewViewModel
+            if (recyclerView != null)
             {
-                ViewModel.CompleteGroups.CollectionChanged += AdjustRecyclerViewHeight;
+                recyclerView.SetItemAnimator(null);
+                recyclerView.OverScrollMode = OverScrollMode.Never;
             }
+
+            SetupTabs();
+            RegisterPageChangeCallback();
+            viewPager.Post(RecalculateRecyclerViewHeight);
+
+            return view;
         }
 
-        private int MeasureItemHeight(View view)
+        private void RegisterPageChangeCallback()
         {
-            view.Measure(
-                View.MeasureSpec.MakeMeasureSpec(recyclerView.Width, MeasureSpecMode.Exactly),
-                View.MeasureSpec.MakeMeasureSpec(0, MeasureSpecMode.Unspecified));
-            return view.MeasuredHeight;
+            if (viewPager == null) return;
+            pageChangeCallback = new CompleteInterviewPageChangeCallback(this);
+            viewPager.RegisterOnPageChangeCallback(pageChangeCallback);
+        }
+
+        internal void OnPageChangedFromCallback()
+        {
+            UpdateTabIndicator();
+            viewPager?.Post(RecalculateRecyclerViewHeight);
+        }
+
+        public class TabConfigurationStrategy : Java.Lang.Object, TabLayoutMediator.ITabConfigurationStrategy
+        {
+            private readonly IMvxAndroidBindingContext bindingContext;
+            private readonly IList<TabViewModel> tabs;
+
+            public TabConfigurationStrategy(IMvxAndroidBindingContext bindingContext, IList<TabViewModel> tabs)
+            {
+                this.bindingContext = bindingContext;
+                this.tabs = tabs;
+            }
+
+            public void OnConfigureTab(TabLayout.Tab tab, int position)
+            {
+                var vm = tabs[position];
+                
+                var view = bindingContext.BindingInflate(Resource.Layout.interview_complete_tab_item, null);
+
+                var countView = view.FindViewById<TextView>(Resource.Id.tab_count);
+                var titleView = view.FindViewById<TextView>(Resource.Id.tab_title);
+                var indicator = view.FindViewById<View>(Resource.Id.tab_indicator);
+
+                countView.Text = vm.Count; 
+                titleView.Text = vm.Title; 
+                tab.View.Enabled = vm.IsEnabled;
+
+                var colorResInt = TabContentToColorConverter.GetColor(vm);
+                var colorInt = ContextCompat.GetColor(tab.View.Context, colorResInt);
+                var color = new Android.Graphics.Color(colorInt);
+                countView?.SetTextColor(color);
+                titleView?.SetTextColor(color);
+                indicator.BackgroundTintList = ColorStateList.ValueOf(color);
+
+                tab.SetCustomView(view);
+                
+                // Subscribe to property changes to update tab state dynamically
+                vm.PropertyChanged += (sender, args) =>

```

Comment:
Never unsubscribes.
Introduce
private readonly Dictionary<TabViewModel, PropertyChangedEventHandler> handlers = new();
Save values, clean up in dispose.